### PR TITLE
raft: Introduce CommittedEntries pagination

### PR DESCRIPTION
**Pull request by bdarnell**
The MaxSizePerMsg setting is now used to limit the size of
Ready.CommittedEntries. This prevents out-of-memory errors if the raft
log has become very large and commits all at once.

I just used the same size limit as we have for MsgApp, although I'd be willing to make it a separate setting if you think it would be useful to configure them separately.

### CODE REVIEW

File: raft/log.go [hunk 1]
```diff
@@ -38,17 +38,20 @@ type raftLog struct {
...
 
 	logger Logger
+
+	maxMsgSize uint64
 }
 
 // newLog returns log using the given storage. It recovers the log to the state
 // that it just commits and applies the latest snapshot.
-func newLog(storage Storage, logger Logger) *raftLog {
+func newLog(storage Storage, logger Logger, maxMsgSize uint64) *raftLog {
```
Comment (siddontang) [inline raft/log.go]: maybe provide another function `newLogWithSize`, so we don't need to change origin newLog in many places?

Comment (bdarnell) [reply raft/log.go]: Done.

**Comment by bdarnell** [comment]
Any more comments @siddontang ?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**